docs/building-a-form/creating-a-form-config-file.md

```diff
@@ -1,286 +1,173 @@
-    expandUnderCondition: 'someValue',
-    // `expandUnderCondition` can also be a function that receives the data from the `expandUnder` field as an argument.
-    expandUnderCondition: (field) => field === 'someValue' || field === 'someOtherValue',
-
-    // When using the expandUnder option, you can set `expandUnderClassNames` on the field specified by `expandUnder` and it will add
-    // classes to the `div` that wraps all of the fields when they're expanded. See cookbook for an example use case.
-    expandUnderClassNames: '',
+  }
+}
+```
 
-    // Hides the specified field on the review page.
-    hideOnReview: true || false,
+In this example:
 
-    // Hides the specified field on the review page when the field value is `false`.
-    hideOnReviewIfFalse: true || false
+1. The user types 'a'.
+2. The `TextWidget` for field1 calls `onChange` with 'a'.
+3. The `onChange` property came from the parent `ObjectField` component, which puts 'a' in an object as `field1` (`{ field1: 'a' }`), then calls the `onChange` prop it was passed.
+4. When it reaches the top-level `Form` component, rjsf runs the JSON Schema validation and passes the results through the component hierarchy.
 
-    // A function that conditionally hides fields in the form. `hideIf` provides the `index` argument when you use `ui:required` on data inside an array.
-    hideIf: function (formData, index) {
-      return true || false;
-    }
+Similar to Redux, all state is kept in the `Form` component, or the root of the form. All data processing and validation happens in `Form`, or is triggered by hooks provided by `Form`. The us-forms-system code built on top of this processes the schemas and form data in Redux, triggered by events provided by `Form`.
 
-    // A function that conditionally replaces the current field's schema. `updateSchema` provides the `index` argument when you use `ui:required` on data inside an array.
-    updateSchema: function (formData, schema, uiSchema, index, pathToCurrentData) {
-      // This function returns an object with the properties you want to update. Instead of replacing the existing schema, it updates the individual properties.
-      return {
-        type: 'string'
-      };
-    },
+##### About the `uiSchema` field component
 
-    // Use this when you have an array field that should not be pulled out of the page its in and shown separately on the review page.
-    keepInPageOnReview: true
-  }
-}
-```
+Along with the regular JSON Schema, a UI schema for UI-specific options that don't fit within the JSON Schema standard is also optionally defined for each field. The `uiSchema` field component is primarily used for specifying custom fields and widgets for specific fields in the schema using `ui:field` and `ui:widget`. In the us-forms-system, it's also used for label names, custom validation, and conditionally required fields.
```

Comment:
This description doesn't really make sense to me. Is this trying to explain the concept of `uiSchema` and what it's used for? I think using the term "field component" is also inaccurate here. I would rewrite this section in the following way:

"Along with the regular JSON Schema, a UI schema for UI-specific options that don't fit within the JSON Schema standard is also optionally defined for each field. The UI schema is passed to the form config file as an object, `uiSchema`, for each field. For more information about what goes into the `uiSchema` object, see here." (with a link to the page about schema and uiSchema)

Reply:
This came from here originally: https://github.com/usds/us-forms-system/pull/112/files#diff-efafc7bc7829dc0c26ee1f6b58608083L230

> Along with the regular JSON Schema, a ui schema is also optionally defined for each field. This schema is used for ui specific options that don't fit within the JSON Schema standard. The primary use for it that's built into the library is for specifying custom fields and widgets for specific fields in the schema (using `ui:field` and `ui:widget`). We've extended it so that we also use it for label names, custom validation, and conditionally required fields.

All that is to say, I think your revision is much more clear than the original and my attempt at a revision. 👍